app/controllers/api/account.php

```diff
@@ -1634,6 +1634,9 @@
         }
 
         $user
+            ->setAttribute('password', $isAnonymousUser ? Auth::passwordHash($password, Auth::DEFAULT_ALGO, Auth::DEFAULT_ALGO_OPTIONS) : $user->getAttribute('password', ''))
+            ->setAttribute('hash', $isAnonymousUser ? Auth::DEFAULT_ALGO : $user->getAttribute('hash', ''))
+            ->setAttribute('hashOptions', $isAnonymousUser ? Auth::DEFAULT_ALGO_OPTIONS : $user->getAttribute('hashOptions', ''))
```

Comment:
@Meldiron is this logic right? Is it supposed to be `!$isAnonymousUser`?

Reply:
Honestly not sure why we ask for password in update phone endpoint. What password is expected in case of anonymous user? Why does "update phone" also update password? If anonymous user used this endpoint and entered password, do I understand correctly this would claim account for him not only by phone, but also by password?

If all of that is expected, then the logic makes sense. If anonymous, generate password hash. If anonymous, configure hash algo. If not, leave as it was.

Reply:
@Meldiron,

> Honestly not sure why we ask for password in update phone endpoint.

Probably as an extra security measure. Same idea behind asking for password for update email. However, the user is less likely to have a password in this case...

> Why does "update phone" also update password?

Probably because update email does. And I think the justification was we want to rehash as soon as we have the password to make sure it's hashed using the default algorithm

> If anonymous user used this endpoint and entered password, do I understand correctly this would claim account for him not only by phone, but also by password?

What do you mean by claim? I think it would set the password, just like the update email API

> If all of that is expected, then the logic makes sense. If anonymous, generate password hash. If anonymous, configure hash algo. If not, leave as it was.

I'd like to get your insight into what should happen since you were involved in adding this type of code to the update email endpoint.

Reply:
@stnguyen90 Any update, are we going to merge this one?

Reply:
Sorry, looks like we're going to change the behavior a bit. 

1. We don't want to update the password for users without a password
2. We don't want to verify the password if the user doesn't have a password